### ISSUE TITLE
Odd behaviour running tests

### ISSUE THREAD

**Issue by russel** [ceylon/ceylon-ide-eclipse#1218]
In a project that compiles successfully (both sources and tests), if I use the pop-up menu on the test module entry in the Ceylon Explorer and select Run As|Ceylon Test, then all the tests execute fine. If I select a single test source file within that module, I can select Run As|Ceylon Test but I just get an exception:

ceylon.language.Exception "invalid type literal: function uk.org.winder.maths_test::factorial_iterative_zero (allowed syntax is prefix function, class, package or module followed with qualified name of declaration)"
	at ceylon.test.core.findCandidatesInTypeLiteral_.findCandidatesInTypeLiteral(DefaultTestRunner.ceylon:253)
	at ceylon.test.core.findCandidates_.findCandidates(DefaultTestRunner.ceylon:183)
	at ceylon.test.core.createExecutors_.createExecutors(DefaultTestRunner.ceylon:85)
	at ceylon.test.core.DefaultTestRunner.initExecutors$priv$(DefaultTestRunner.ceylon:43)
	at ceylon.test.core.DefaultTestRunner.<init>(DefaultTestRunner.ceylon:46)
	at ceylon.test.createTestRunner_.createTestRunner(TestRunner.ceylon:49)
	at ceylon.test.createTestRunner_.createTestRunner(TestRunner.ceylon)
	at com.redhat.ceylon.testjvm.Runner.run(tool.ceylon:87)
	at com.redhat.ceylon.testjvm.run_.run(tool.ceylon:39)
	at com.redhat.ceylon.testjvm.run_.main(tool.ceylon)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at ceylon.modules.api.runtime.SecurityActions.invokeRunInternal(SecurityActions.java:58)
	at ceylon.modules.api.runtime.SecurityActions.invokeRun(SecurityActions.java:48)
	at ceylon.modules.api.runtime.AbstractRuntime.invokeRun(AbstractRuntime.java:101)
	at ceylon.modules.api.runtime.AbstractRuntime.execute(AbstractRuntime.java:167)
	at ceylon.modules.api.runtime.AbstractRuntime.execute(AbstractRuntime.java:151)
	at ceylon.modules.Main.execute(Main.java:69)
	at ceylon.modules.Main.main(Main.java:42)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.jboss.modules.Module.run(Module.java:312)
	at org.jboss.modules.Main.main(Main.java:460)
	at ceylon.modules.bootstrap.CeylonRunTool.run(CeylonRunTool.java:208)
	at ceylon.modules.bootstrap.CeylonTestTool.run(CeylonTestTool.java:172)
	at com.redhat.ceylon.common.tools.CeylonTool.run(CeylonTool.java:352)
	at com.redhat.ceylon.common.tools.CeylonTool.execute(CeylonTool.java:291)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at com.redhat.ceylon.launcher.Launcher.runInJava7Checked(Launcher.java:104)
	at com.redhat.ceylon.launcher.Launcher.run(Launcher.java:37)
	at com.redhat.ceylon.launcher.Launcher.run(Launcher.java:30)
	at com.redhat.ceylon.launcher.Launcher.main(Launcher.java:23)

**Comment by gavinking** [comment]
@russel Does this package belong to a module, or is it in a package of the default module? What is the name of the module?

**Comment by russel** [comment]
Sorry I should have provided better detail, poor reporting on my part. The project is just 4 implementations of factorial with associated test (well it will be soon, currently there are just 4 tests to get things started).  The project structure is:
```
├── source
│   └── uk
│       └── org
│           └── winder
│               └── maths
│                   ├── factorial.ceylon
│                   ├── module.ceylon
│                   └── package.ceylon
└── test-source
    └── uk
        └── org
            └── winder
                └── maths_test
                    ├── factorial.ceylon
                    ├── module.ceylon
                    └── package.ceylon
```
source/uk/org/winder/maths/module.ceylon:
```Ceylon
module uk.org.winder.maths "0.0.0" {
  import java.base "8";
}
```
source/uk/org/winder/maths/package.ceylon:
```Ceylon
shared package uk.org.winder.maths;
```
source/uk/org/winder/maths/module.ceylon:
```Ceylon
module uk.org.winder.maths_test "0.0.0" {
  import ceylon.test "1.1.0";
  import uk.org.winder.maths "0.0.0";
  import java.base "8";
}
```
source/uk/org/winder/maths/package.ceylon:
```Ceylon
package uk.org.winder.maths_test;
```

**Comment by gavinking** [comment]
@russel thanks.

**Comment by thradec** [comment]
What version of IDE and compiler do you use? (official 1.1 or something newer)

**Comment by thradec** [comment]
Found it, fixed via ceylon/ceylon-sdk@e00db081a. Thanks for reporting.
